app/controllers/request-to-go-live/index/get.controller.js

```diff
@@ -1,16 +1,82 @@
 'use strict'
 
+// NPM dependencies
+const lodash = require('lodash')
+
 // Local dependencies
 const goLiveStage = require('../../../models/go-live-stage')
 const response = require('../../../utils/response')
 
+const {
+  NOT_STARTED,
+  ENTERED_ORGANISATION_NAME,
+  CHOSEN_PSP_STRIPE,
+  CHOSEN_PSP_WORLDPAY,
+  CHOSEN_PSP_SMARTPAY,
+  CHOSEN_PSP_EPDQ,
+  TERMS_AGREED_STRIPE,
+  TERMS_AGREED_WORLDPAY,
+  TERMS_AGREED_SMARTPAY,
+  TERMS_AGREED_EPDQ,
+  LIVE,
+  DENIED
+} = goLiveStage
+
+const live = [
+  LIVE
+]
+
+const agreedToTerms = [
+  TERMS_AGREED_STRIPE,
+  TERMS_AGREED_WORLDPAY,
+  TERMS_AGREED_SMARTPAY,
+  TERMS_AGREED_EPDQ
+].concat(live)
+
+const chosenHowToProcessPayments = [
```

Comment:
(This certainly shouldn't be blocking - it depends on how often this pattern will continue in the future. tl;dr define constants and groups once)

It seems like we're pulling and categorising these constant out of this object map in a number of different controllers. Would it make sense to move this logic over to the helper object/ model? 

Potential pros: 
* controllers don't have to duplicate code 
* controllers are simple and easy to read (although this code is very clear!) 

Something like: 
```js
const goLiveStage = require('goLiveStage')

const goLiveIndexHandler = function goLiveIndexHandler (req, res, next) { 
  const { currentGoLiveStage } = req.service
  const goLiveProcessStatus = goLiveStage.status(currentGoLiveStage)

  // ensure cookies set in previous session aren't propagated
  delete req.session.pageData.requestToGoLive.index

  return response(req, res, 'request-to-go-live/index', pageData)
}

module.exports = goLiveIndexHandler
```

Or  a middle ground would be to do the categorising only once in the helper/ model and then use those for logic in the controllers: 

```js
const { agreedToTermsStates, sessionIsAgreeingToTerms } = require('goLiveStage') 

// check if my status is in the agreed terms
agreedToTermsStates.includes(currentGoLiveStage) 

// or 
sessionIsAgreeingToTerms(currentGoLiveStage)
```

Files that also do this: 
https://github.com/alphagov/pay-selfservice/blob/master/app/controllers/request-to-go-live/choose-how-to-process-payments/post.controller.js#L16
https://github.com/alphagov/pay-selfservice/blob/master/app/controllers/request-to-go-live/agreement/get.controller.js#L12

Reply:
Yes, something to consider for the future